### PR TITLE
mfcl5750dw: init at 3.5.1-1

### PR DESCRIPTION
**Pull request by qdlmcfresh**
Add CUPS driver for Brother MFC-L5750DW printer

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by qdlmcfresh** [comment]
Result of nixpkgs-review pr 277781 run on x86_64-linux:
```
1 package built:
mfcl5750dw
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please move this into `pkgs/by-name`.

**Comment by qdlmcfresh** [comment]
> Please move this into `pkgs/by-name`.

Ok im happy to do that, but any comment on why?

**Comment by Pandapip1** [comment]
> Ok im happy to do that, but any comment on why?

It's now preferred over `pkgs/misc`.

**Comment by qdlmcfresh** [comment]
> Please move this into `pkgs/by-name`.

i moved the package and rebased against master

File: pkgs/by-name/mf/mfcl5750dw/package.nix [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  lpr_src = fetchurl {
+    url = "https://download.brother.com/welcome/dlf102613/mfcl5750dwlpr-${version}.i386.deb";
+    sha256 = "b1b9c1f9ae8522a65fdd2db1e760aed835807ba593d001ac5471635a157cd1f1";
+  };
+
+  unpackPhase = ''
+    dpkg-deb -x $src .
+    dpkg-deb -x $lpr_src .
+  '';
```
Comment (Pandapip1) [inline pkgs/by-name/mf/mfcl5750dw/package.nix:57]:
```suggestion
  srcs = [
    (fetchurl {
      name = "mfcl5750dwcupswrapper";
      url = "https://download.brother.com/welcome/dlf102614/mfcl5750dwcupswrapper-${version}.i386.deb";
      sha256 = "afe6d18e17d26348f3b8a4f9a003107984940c429a7dd193054303a21f5e65b5";
    })
    (fetchurl {
      name = "mfcl5750dwlpr";
      url = "https://download.brother.com/welcome/dlf102613/mfcl5750dwlpr-${version}.i386.deb";
      sha256 = "b1b9c1f9ae8522a65fdd2db1e760aed835807ba593d001ac5471635a157cd1f1";
    })
  ];
  unpackPhase = ''
    runHook preUnpack
    for src in $srcs; do
      dpkg-deb -x $src .
    done
    runHook postUnpack
  '';
```

File: pkgs/by-name/mf/mfcl5750dw/package.nix [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  nativeBuildInputs = [
+    dpkg
+    makeWrapper
+    autoPatchelfHook
+  ];
+  buildInputs = [ perl ];
+
+  src = fetchurl {
+    url = "https://download.brother.com/welcome/dlf102614/mfcl5750dwcupswrapper-${version}.i386.deb";
```
Comment (Pandapip1) [inline pkgs/by-name/mf/mfcl5750dw/package.nix:45]: Can I confirm that fetching the `i386` version is correct for all the arches?

File: pkgs/by-name/mf/mfcl5750dw/package.nix [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+}:
+
+let
+  arches = [
+    "x86_64"
+    "i686"
+    "armv7l"
+  ];
+
+  runtimeDeps = [
```
Comment (Pandapip1) [inline pkgs/by-name/mf/mfcl5750dw/package.nix:24]: This list should be added to `buildInputs`, and `buildInputs` should be used instead.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mf/mfcl5750dw/package.nix [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      ln -s \
+      $out/opt/brother/Printers/MFCL5750DW/cupswrapper/lpdwrapper \
+      $out/lib/cups/filter/brother_lpdwrapper_MFCL5750DW
+      ln -s \
+      $out/opt/brother/Printers/MFCL5750DW/cupswrapper/brother-MFCL5750DW-cups-en.ppd \
+      $out/share/cups/model/
+      runHook postInstall
+    '';
+  meta = with lib; {
+    homepage = "http://www.brother.com/";
```
Comment (Pandapip1) [inline pkgs/by-name/mf/mfcl5750dw/package.nix:114]:
```suggestion
    homepage = "https://www.brother.com/";
```

**Review by Pandapip1 — APPROVED** [review]
While I'm not particularly happy with the way the derivation is organized, it doesn't require any changes.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**